tests/test_routes.py

```diff
@@ -272,6 +264,113 @@ def test_route_fund_dashboard_clear_filters(
+        self,
+        request,
+        flask_test_client,
+        mock_get_fund,
+        mock_get_round,
+        mock_get_application_overviews,
+        mock_get_assessment_stats,
+        mock_get_assessment_progress,
+        sort_column,
+        sort_order,
+        column_id,
+    ):
+
+        params = request.node.get_closest_marker("mock_parameters").args[0]
+        fund_short_name = params["fund_short_name"]
+        round_short_name = params["round_short_name"]
+
+        response = flask_test_client.get(
+            f"/assess/assessor_dashboard/{fund_short_name}/{round_short_name}",
+            follow_redirects=True,
+            query_string={
+                "sort_column": sort_column,
+                "sort_order": sort_order,
+            },
+        )
+
+        assert 200 == response.status_code, "Wrong status code on response"
+        soup = BeautifulSoup(response.data, "html.parser")
+
+        # Find the table element by its class name
+        tbody = soup.find("tbody", {"class": "govuk-table__body"})
+
+        # Find all the elements in the column
+        column_data = [
+            row.find_all("td")[column_id].text
+            for idx, row in enumerate(tbody.find_all("tr"))
+        ]
+
+        if sort_order == "asc":
+            all_table_data_elements = str(
+                soup.find_all(
+                    "th",
+                    attrs={
+                        "class": "govuk-table__header",
+                        "aria-sort": "ascending",
+                    },
+                )
+            )
+            assert 'aria-sort="ascending"' in all_table_data_elements
+            assert sort_column in all_table_data_elements
+            # check if the data is in ascending order
+            assert all(
+                column_data[i] <= column_data[i + 1]
+                for i in range(len(column_data) - 1)
+            )
+        elif sort_order == "desc":
+            all_table_data_elements = str(
+                soup.find_all(
+                    "th",
+                    attrs={
+                        "class": "govuk-table__header",
+                        "aria-sort": "descending",
+                    },
+                )
+            )
+            assert 'aria-sort="descending"' in all_table_data_elements
+            assert sort_column in all_table_data_elements
+            # check if the data is in descending order
+            assert all(
+                column_data[i] >= column_data[i + 1]
+                for i in range(len(column_data) - 1)
+            )
+        else:
+            all_table_data_elements = str(
+                soup.find_all(
+                    "th",
+                    attrs={
+                        "class": "govuk-table__header",
+                        "aria-sort": "none",
+                    },
+                )
+            )
+            assert 'aria-sort="none"' in all_table_data_elements
```

Comment:
I feel these should be 3 separate tests, rather than using parametrize & branched conditions, what do you think?

Reply:
Here I'm testing the column sorting on multiple columns(`location` & `funding`). So I think parametrization makes more sense than splitting into multiple test functions